opal/mca/hwloc/external/configure.m4

```diff
@@ -146,6 +143,34 @@ AC_DEFUN([MCA_opal_hwloc_external_CONFIG],[
                   AC_CHECK_DECLS([HWLOC_OBJ_OSDEV_COPROC], [], [], [#include <hwloc.h>])
                   AC_CHECK_FUNCS([hwloc_topology_dup])])
 
+           AS_IF([test "$opal_hwloc_external_support" = "yes"],
+                 [AC_MSG_CHECKING([if external hwloc version is 1.5 or greater])
+                  AC_COMPILE_IFELSE(
+                      [AC_LANG_PROGRAM([[#include <hwloc.h>]],
+                          [[
+#if HWLOC_API_VERSION < 0x00010500
+#error "hwloc API version is less than 0x00010500"
+#endif
+                          ]])],
+                      [AC_MSG_RESULT([yes])],
+                      [AC_MSG_RESULT([no])
+                       opal_hwloc_external_support=no])])
+
+           AS_IF([test "$opal_hwloc_external_support" = "yes" &&
+                  test -z "$with_hwloc"],
+                 [AC_MSG_CHECKING([if external hwloc version is 2.0 or greater])
+                  AC_COMPILE_IFELSE(
+                      [AC_LANG_PROGRAM([[#include <hwloc.h>]],
+                          [[
+#if HWLOC_API_VERSION < 0x00020000
+#error "hwloc API version is less than 0x00020000"
```

Comment:
Is there a way to have this version by reference instead of by value?  I.e., I imagine that this will be one more place to forget to update if/when we update the embedded version of hwloc.

Reply:
I do not know ... strictly speaking, this is not even the library version (e.g. 2.0.1) but the API version (e.g. 2.0.0).

@bgoglin is there any way to test the __library__version instead of the API version at compile time ?
for example, `libevent` defines `_EVENT_NUMERIC_VERSION`.

Reply:
@ggouaillardet Don't you also need to test that ```$with_hwloc``` wasn't given without any path (and so has a ```yes``` value)? I thought that ```test -z``` just means it wasn't given at all.

Reply:
@rhc54 you are right, and I just pushed a commit that fixes that.